pennylane/devices/experimental/device_api.py

```diff
@@ -168,6 +168,16 @@ def __init__(self, wires=None, shots=None) -> None:
 
         self._wires = wires
 
+    def __repr__(self):
+        """String representation."""
+        details = []
+        if self.wires:
+            details.append(f"wires={len(self.wires)}")
+        if self.shots:
+            details.append(f"shots={self.shots.total_shots}")
+        details = f"({', '.join(details)}) " if details else ""
+        return f"<{self.__class__.__name__} device {details}at {hex(id(self))}>"
```

Comment:
```suggestion
        return f"<{self.name} device {details}at {hex(id(self))}>"
```

Maybe?

Also, it might be worth including `max_workers` in the repr for `DefaultQubit2`.

Reply:
Any purpose to still including the `hex(id(self))`?

Reply:
I figured it would help distinguish between identical devices, but I'm now realizing that the new devices are (or at least, should be) stateless. I still think it wouldn't hurt. perhaps one's RNG has a different seed, or has been called a different amount of times? you (I) never know what you (I) might miss

Reply:
just realizing max_workers is only for DQ2. I don't really want to maintain 2 repr methods just to add one thing... maybe if we change it in the future, but for now i'm gonna leave it out